src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4288,7 +4288,9 @@ public bool SendToAsync(SocketAsyncEventArgs e)
             }
             if (e.RemoteEndPoint == null)
             {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly, argument not applicable
                 throw new ArgumentNullException(nameof(RemoteEndPoint));
+#pragma warning restore CA2208
```

Comment:
There is a test
```csharp
[Fact]
public async Task NullEndpoint_Throws()
{
    using var socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
    await Assert.ThrowsAsync<ArgumentNullException>(() => SendToAsync(socket, new byte[1], null));
}
```
which ` SendToAsync(socket, new byte[1], null));` calls this method using TestHelper overloads. As this method haven't `remoteEndPoint` argument compiler warning here, but changing this fails the test, changing the test fails the other overload tests. All other overloads/methods called by TestHelper has that arguments and throws ArgumentNullException correctly. This makes me think somehow this exceptions supposed to be same, so want to  check with area owners. @davidsh @scalablecory is it OK to change this exception with `ArgumentException(SR.Format(SR.InvalidNullArgument, "e.RemoteEndPoint"), nameof(e));` and update corresponding tests?

Reply:
I'm not following.  What is the failure you get when you fix this exception to use `nameof(e)` instead of `nameof(RemoteEndPoint)`?

Reply:
I should have changed it with `throw new ArgumentException(SR.Format(SR.InvalidNullArgument, "e.RemoteEndPoint"), nameof(e));`

But above test failing with error `Expected ArgumentNullException, found ArgumentException`  because 2nd row of the test `SendToAsync(socket, new byte[1], n)` calling this method by TestHelper overloads. The question is if for any reason the exceptions of SendToAsync(...) overloads expected to be same? If not i can update the test helper to not call at least `SendToAsync(SocketAsyncEventArgs e)`, hopefully that wouldn't break other tests

Reply:
fixed the warning and updated the test